core/src/main/java/com/scalar/db/api/ScanAll.java

```diff
@@ -24,26 +24,61 @@ public ScanAll(ScanAll scanAll) {
     super(scanAll);
   }
 
+  /**
+   * Guaranteed to throw an exception
+   *
+   * @throws UnsupportedOperationException always
+   * @deprecated Unsupported operation.
+   */
+  @Deprecated
   @Override
   public ScanAll withStart(Key clusteringKey) {
     throw new UnsupportedOperationException();
   }
```

Comment:
By adding the `@Deprecated` annotation on methods that throw an `UnsupportedOperationException.`, the user IDE will show a warning. 
We hope it will catch the user's attention so that he does try to use these methods.